langchain/document_loaders/json_loader.py

```diff
@@ -87,17 +87,17 @@ def load(self) -> List[Document]:
             )
 
             if self._content_key is not None:
-                text = sample.get(self._content_key)
+                content = sample.get(self._content_key)
```

Comment:
is something else going wrong if `sample.get(self._content_key)` is a dict (instead of str | None)? cc @avsolatorio

Reply:
Hello @triet-lq-holistics , right! The use case I had in mind when implementing the JSONLoader was that only fields containing text are considered valid. But some use cases may very well require storing a full subset of the JSON data, and I think we should support that.

However, I think that the behavior of converting arbitrary non-text data as input to the `page_content` should be made explicit by the user. For example, exposing an argument like `text_content: bool = True` that must be checked in the initial set of validators could be useful.

```Python
if text_content and not isinstance(content, str):
    raise ValueError(f"Expected page_content is string, got {type(content)} instead. Set `text_content=False` if the desired input for `page_content` is not a string.")
````

The reason why this could be useful is because `jq` has a very permissive syntax. And so, a user may unintentionally store non-text content if we don't have this in place.

wdyt, @dev2049 ?

Reply:
im a fan of that idea @avsolatorio!